lib/spack/spack/compilers/__init__.py

```diff
@@ -176,18 +185,55 @@ def all_compiler_specs(scope=None, init_config=True):
             for s in all_compilers_config(scope, init_config)]
 
 
-def find_compilers(*paths):
-    """Return a list of compilers found in the supplied paths.
-       This invokes the find_compilers() method for each operating
-       system associated with the host platform, and appends
-       the compilers detected to a list.
+def find_compilers(path_hints=None):
+    """Returns the list of compilers found in the paths given as arguments.
+
+    Args:
+        path_hints (list or None): list of path hints where to look for.
+            A sensible default based on the ``PATH`` environment variable
+            will be used if the value is None
+
+    Returns:
+        List of compilers found
     """
-    # Find compilers for each operating system class
-    oss = all_os_classes()
-    compiler_lists = []
-    for o in oss:
-        compiler_lists.extend(o.find_compilers(*paths))
-    return compiler_lists
+    if path_hints is None:
```

Comment:
@alalazo I think this check might be overly strict. I'm seeing cases where `path_hints` is `[]` for `spack compiler find` (which I expect would check `PATH`) and so this check is not triggered.